app/lib/actions/remote_execution/run_host_job.rb

```diff
@@ -47,10 +48,17 @@ def plan(job_invocation, host, template_invocation, proxy_selector = ::RemoteExe
         raise _('Failed rendering template: %s') % renderer.error_message unless script
 
         provider = template_invocation.template.provider
-        hostname = provider.find_ip_or_hostname(host)
+
+        secrets = { :ssh_password => job_invocation.password || provider.ssh_password(host),
+                    :key_passphrase => job_invocation.key_passphrase || provider.ssh_key_passphrase(host) }
```

Comment:
This gives me `Failed to initialize: Dynflow::ExecutionPlan::Steps::Error - private method `ssh_password' called for SSHExecutionProvider:Class`